### PR TITLE
Address some PR comments

### PR DESCRIPTION
**Pull request by emmberk**
@jhunzik

### CODE REVIEW

File: src/main/java/com/connexta/store/controllers/StoreController.java [hunk 1]
```diff
@@ -120,24 +158,29 @@ public StoreController(@NotNull final StoreService storeService) {
 
       log.warn("Unable to retrieve {}", productId, e);
       return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
-    } catch (Exception exception) {
+    } catch (Throwable t) {
```
Comment (ahoffer) [inline src/main/java/com/connexta/store/controllers/StoreController.java]: :exclamation:  Throwable will catch thing we don't want to catch like JVM errors and out-of-memory exceptions.

File: src/main/java/com/connexta/store/controllers/StoreController.java [hunk 2]
```diff
@@ -91,8 +92,45 @@ public StoreController(@NotNull final StoreService storeService) {
   }
 
   @Override
-  public @NotNull ResponseEntity<Resource> retrieveProduct(String productId) {
-    new UuidValidator().validate(productId);
+  public ResponseEntity<Void> addMetadata(
+      final String acceptVersion,
+      @Pattern(regexp = "^[0-9a-zA-Z]+$") @Size(min = 32, max = 32) final String productId,
+      @Pattern(regexp = "^[0-9a-zA-Z\\-]+$") @Size(min = 1, max = 32) final String metadataType,
```
Comment (ahoffer) [inline src/main/java/com/connexta/store/controllers/StoreController.java]: :exclamation: The regex `^[0-9a-zA-Z\\-]+$` allows product IDs with back slashes. E.g. `aa\bb-cc\`

File: src/main/java/com/connexta/store/controllers/StoreController.java (hunk 1 above)
Comment (emmberk) [reply src/main/java/com/connexta/store/controllers/StoreController.java]: Yes. I few lines down, it gets re-thrown.

File: src/main/java/com/connexta/store/controllers/StoreController.java (hunk 2 above)
Comment (emmberk) [reply src/main/java/com/connexta/store/controllers/StoreController.java]: Whoops. I meant to allow dashes, not slashes. The fix needs to be made in the API spec, not this repo.

File: src/main/java/com/connexta/store/controllers/StoreController.java (hunk 1 above)
Comment (ahoffer) [reply src/main/java/com/connexta/store/controllers/StoreController.java]: :man_facepalming:

**Pull request merged**